### PR TITLE
Stable support for MSC4115-style membership properties

### PR DESCRIPTION
**Pull request by richvdh**
MSC4115 has now completed FCP, so we can move it to the main test suite and use stable properties.

Requires https://github.com/element-hq/synapse/pull/17282 to land before it will pass CI.

### CODE REVIEW

**Comment by richvdh** [comment]
> Requires https://github.com/element-hq/synapse/pull/17282 to land before it will pass CI.

Oh, no it doesn't, because complement's CI checks out the matching branch of Synapse.

File: tests/csapi/membership_on_events_test.go [hunk 1]
```diff
@@ -10,11 +10,11 @@ import (
 	"github.com/matrix-org/complement/helpers"
 )
 
-// MSC4115: membership information on events
+// Membership information on events served to clients, as specified in MSC4115.
 //
-// Alice sends one message before bob joins, then one after. Bob reads both messages, and checks the membership state
+// Alice sends one message before Vob joins, then one after. Bob reads both messages, and checks the membership state
```
Comment (reivilibre) [inline tests/csapi/membership_on_events_test.go:15]: Vob? :D

**Review by reivilibre — APPROVED** [review]
fine otherwise :D

File: tests/csapi/membership_on_events_test.go (hunk 1 above)
Comment (richvdh) [reply tests/csapi/membership_on_events_test.go:15]: :facepalm: 
```suggestion
// Alice sends one message before Bob joins, then one after. Bob reads both messages, and checks the membership state
```

**Pull request merged**